plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/DeployImageWizardModel.java

```diff
@@ -201,8 +201,12 @@ public void initModel(final Connection connection, final IProject project, boole
 				protected IStatus doRun(IProgressMonitor monitor) {
 					try {
 						List<IProject> projects = connection.getResources(ResourceKind.PROJECT);
+						//Project might be set by a master model that set connection to this slave model
+						//and thus caused loading projects job. While this job was in wait,
+						//master model might set project. It must be respected.
```

Comment:
I described in details the race in comment to https://issues.jboss.org/browse/JBIDE-23210.
The goal of this fix is to provide the same result whatever is the winner in the race.
If wizard in main thread sets project to this model before loading job is completed (higher probability), the job will respect the the project by calling getProject().
If Eclipse gives the loading job priority and main will wait until its completion (lower probability), then main will set the project as it was expected.